src/partialcor.jl

```diff
@@ -0,0 +1,61 @@
+# Partial correlation
+
+export PartialCorTest
+
+"""
+    PartialCorTest(x, y, Z)
+
+Perform a t-test for the hypothesis that ``\\text{Cor}(x,y|Z=z) = 0``, i.e. the partial
+correlation of vectors `x` and `y` given the matrix `Z` is zero.
+
+Implements `pvalue` and `confint`. See also [`partialcor`](@ref).
+"""
+struct PartialCorTest <: HypothesisTest
+    r::Real
+    n::Int
+    k::Int
+    t::Real
+
+    # Error checking is done in `partialcor`
+    function PartialCorTest(x::AbstractVector, y::AbstractVector, Z::AbstractMatrix)
+        r = partialcor(x, y, Z)
```

Comment:
Is `partialcor` guaranteed in `[0,1]`?

Reply:
Yep. As implemented in StatsBase, it returns the result of `clampcor`.